### ISSUE TITLE
The best way to implement pagination?

### ISSUE THREAD

**Issue by gtamas** [microsoft/etcd3#138]
I know that getAll().all() returns all keys, however if the ETCD server has many keys (thousands or tens of thousands), the query will fail because of resource limits.

What's the best way to implement pagination? I guess MultiRangeBuilder is the way to go and I checked the docs, but I don't see an easy way to do this.

Could you post a basic example? I assume this should be accomplished by using miCreateRevision, maxCreateRevision and limit? Another way might be using prefixes, but not all ETCD keys are prefixed.

Basically I just need a general way to query and list any number of keys via pagination or infinite scroll.

Thanks in advance.

**Comment by connor4312** [comment]
I recommending using [limit](https://microsoft.github.io/etcd3/classes/multirangebuilder.html#limit) and [sort](https://microsoft.github.io/etcd3/classes/multirangebuilder.html#sort) and page through by incrementing [minCreateRevision](https://microsoft.github.io/etcd3/classes/multirangebuilder.html#mincreaterevision). The etcd revision is incremented every time a mutation is made, so this will provide you an easy, duplicate-free (provided no key is deleted and recreated while you page) way to page through data.
